lib/rouge/lexers/javascript.rb

```diff
@@ -97,22 +97,26 @@ def self.analyze_text(text)
 
       def self.keywords
         @keywords ||= Set.new %w(
-          for in while do break return continue switch case default
+          for in of while do break return continue switch case default
           if else throw try catch finally new delete typeof instanceof
-          void this yield
+          void this yield import export from as async super this
         )
       end
 
       def self.declarations
-        @declarations ||= Set.new %w(var let with function)
+        @declarations ||= Set.new %w(
+          var let const with function class
+          extends constructor get set
+        )
       end
 
       def self.reserved
         @reserved ||= Set.new %w(
-          abstract boolean byte char class const debugger double enum
-          export extends final float goto implements import int interface
+          abstract boolean byte char debugger double enum
+          final float goto implements int interface
           long native package private protected public short static
-          super synchronized throws transient volatile
+          synchronized throws transient volatile
+          eval arguments await
```

Comment:
`await` is always a keyword in `async` functions or modules, but free to use as identifier outside them.

`eval` and `arguments` are not keywords but are, nonetheless, always forbidden as identifiers.